server/src/parsing/error-parsing.ts

```diff
@@ -154,79 +154,135 @@ Diff:
-  {pre: /^\s*(?:Error:\s+)?(?:.*\n)*Refiner\s+was\s+given\s+an\s+argument\s+".*?"\s+of\s+type\s+"/, mid: /"\s+instead\s+of\s+"/g, post: /"[.]?$/},
-  {pre: /^\s*(?:Error:\s+)?(?:.*\n)*The\s+file\s+.*?\.vo\s+contains\s+library\s+/, mid: /\s+and\s+not\s+library\s+/, post: /[.]?$/},
-  {pre: /^\s*(?:Error:\s+)?(?:.*\n)*The\s+term\s+.*?\s+has\s+type\s+"/, mid: /"\s+while\s+it\s+is\s+expected\s+to\s+have\s+type\s+"/, post: /"[.]?$/},
+  {
+    pre: /^\s*(?:Error:\s+)?(?:.*\n)*Found\s+target\s+class\s+"/,
+    mid: /"\s+instead\s+of\s+"/g,
+    post: /"[.]?$/
+  },
+  {
+    pre: /^\s*(?:Error:\s+)?(?:.*\n)*Impossible\s+to\s+unify\s+"/,
+    mid: /"\s+with\s+"/g,
+    post: /"[.]?$/
+  },
+  {
+    pre: /^\s*(?:Error:\s+)?(?:.*\n)*Unable\s+to\s+unify\s+"/,
+    mid: /"\s+with\s+"/g,
+    post: /"[.]?$/
+  },
+  {
+    pre: /^\s*(?:Error:\s+)?(?:.*\n)*Refiner\s+was\s+given\s+an\s+argument\s+".*?"\s+of\s+type\s+"/,
+    mid: /"\s+instead\s+of\s+"/g,
+    post: /"[.]?$/
+  },
+  {
+    pre: /^\s*(?:Error:\s+)?(?:.*\n)*The\s+file\s+.*?\.vo\s+contains\s+library\s+/,
+    mid: /\s+and\s+not\s+library\s+/,
+    post: /[.]?$/
+  },
+  {
+    pre: /^\s*(?:Error:\s+)?(?:.*\n)*The\s+term\s+.*?\s+has\s+type\s+"/,
+    mid: /"\s+while\s+it\s+is\s+expected\s+to\s+have\s+type\s+"/,
+    post: /"[.]?$/
+  }
 ];
 
-function* indicesOf(str: string, substr: string|RegExp, start?: number, end?: number) : Iterable<[number,number]> {
-  function indexOf(str: string,substr: string|RegExp, pos: number) : [number,number] {
-    if(typeof substr==='string') {
-      const i = str.indexOf(substr,pos);
-      return [i, i >=0 ? i+substr.length : 0]
+function* indicesOf(
+  str: string,
+  substr: string | RegExp,
+  start?: number,
+  end?: number
+): Iterable<[number, number]> {
+  function indexOf(
+    str: string,
+    substr: string | RegExp,
+    pos: number
+  ): [number, number] {
+    if (typeof substr === "string") {
+      const i = str.indexOf(substr, pos);
+      return [i, i >= 0 ? i + substr.length : 0];
     } else {
       substr.lastIndex = pos;
       const m = substr.exec(str);
-      if(!m || m.index < pos)
-        return [-1,0];
-      else
-        return [m.index, m.index+m[0].length];
+      if (!m || m.index < pos) return [-1, 0];
+      else return [m.index, m.index + m[0].length];
     }
   }
   start = start || 0;
   end = end || start;
   let pos = start;
 
-  let idx = indexOf(str,substr,pos);
-  while(idx[0] >= 0 && pos < end) {
+  let idx = indexOf(str, substr, pos);
+  while (idx[0] >= 0 && pos < end) {
     yield idx;
-    pos = idx[0]+1;
-    idx = indexOf(str,substr,pos);
+    pos = idx[0] + 1;
+    idx = indexOf(str, substr, pos);
   }
 }
 
-function costOfDifference(d: diff.IDiffResult[]) : number {
-  return d.reduce((v,x) => v + (x.added||x.removed ? x.value.length : 0), 0);
+function costOfDifference(d: diff.Change[]): number {
+  return d.length;
```

Comment:
@artagnon is this the same thing as before?

Reply:
Yep, the API changed, and I think I found the exact equivalent: you've got to test it to be sure.

Reply:
This seems very strange: the previous cost model was depending on the length of inserted/removed string. How can that be reflected in the length of the array?

Reply:
TBH, I don't remember much of my reasoning; I wrote this a long time ago. However, there is an API reference [here](https://github.com/kpdecker/jsdiff) that you can use as a guide. I used `diffLines` instead of `diffWords`: that's the main difference I think.